changelogs/fragments/81606-ansible-galaxy-collection-pre-releases.yml

```diff
@@ -0,0 +1,16 @@
+---
+
+bugfixes:
+- >-
+  ansible-galaxy - started allowing the use of pre-releases
+  for dependencies on any level of the dependency tree that
+  specifically demand exact pre-release versions of
+  collections and not version ranges.
+  (https://github.com/ansible/ansible/pull/81606)
+- >-
+  ansible-galaxy - started allowing the use of pre-releases
```

Comment:
Does pip work like this?

Reply:
Yep

Reply:
Retrieved some context for history:
* pip's rationale (which I'd apparently thumbed up years ago): https://github.com/pypa/pip/issues/5503#issuecomment-397426044
* PEP 440: https://peps.python.org/pep-0440/#handling-of-pre-releases

While googling I've also found that another tool — pipenv — might have a problem with this, but not pip itself:
* https://github.com/pypa/pipenv/issues/2022#issuecomment-382706536
* https://github.com/psf/black/issues/2115